include/aspect/utilities.h

```diff
@@ -604,6 +604,25 @@ namespace aspect
          */
         explicit AsciiDataLookup(const double scale_factor);
 
+        /**
+         * Replace the data stored in this class by the data given to this function.
+         *
+         * @note This is a manual way to fill the data. Consider load_file() if your data
+         * is stored in a file.
+         *
+         * The data consists of @p n_components (implicitly given by the size of @p column_names)
+         * specified at @p n_points_per_direction in each of the dim coordinate directions.
+         *
+         * The data in @p raw_data consists of dim+n_components number of tables, where the
+         * first dim tables are the coordinate values (raw_data[0] is the table of x
+         * coordinates, raw_data[1] is the table of y coordinates, etc..) and the remaining
+         * columns are the data columns.
+         */
+        void reinit(const TableIndices<dim> &n_points_per_direction,
+                    const std::vector<std::string> &column_names,
+                    const std::vector<Table<dim,double> > &raw_data
```

Comment:
I don't understand your choice of data format.

If I understand this right, then you have a 3d collection of data points, at each of which you store `n_components` data values. But these 3d points are not arbitrarily distributed: they're either uniformly or non-uniformly spaced along each axis, but they're tensor products of x-, y-, and z-spaced points. (That's necessary because the classes that interpolated are either InterpolatedUniformGridData or InterpolatedTensorProductGridData).

So I don't understand why you would want to store the x, y, and z-coordinates of all points as 3d tables: That's highly redundant. All you need to provide are `dim` vectors with abscissae, and that's also what needs to be input into the two classes mentioned above.

Reply:
> So I don't understand why you would want to store the x, y, and z-coordinates of all points as 3d tables

I did not come up with this. This is extracted from ``load_file`` without changes. I agree that this not great (and commented about this at the top of this PR.

Reply:
Hm, what do others think? I'm a bit torn between saying "that's how it always was" and saying "if you're already rewriting it, let's do this right".